Sources/Brave/Frontend/Browser/TabManager.swift

```diff
@@ -416,12 +416,30 @@ class TabManager: NSObject {
     }
     // When bulk adding tabs don't notify delegates until we are done
     self.isRestoring = true
-    var tab: Tab!
-    for url in urls {
-      tab = self.addTab(URLRequest(url: url), flushToDisk: false, zombie: zombie, isPrivate: isPrivate)
+    let tabs = urls.map {
+      let tab = self.addTab(URLRequest(url: $0),
+                            flushToDisk: false,
+                            zombie: true,
+                            isPrivate: isPrivate)
+      tab.lastTitle = $0.absoluteDisplayString
+      tab.url = $0
+      tab.favicon = FaviconFetcher.getIconFromCache(for: $0) ?? Favicon.default
```

Comment:
Should we lazy-load favicons as well?

Reply:
I found it weird to not show them. Reason is because then I see blank tabs with blank fav-icons (the globe/default icon)